pex/commands/command.py

```diff
@@ -88,6 +84,53 @@ def __init__(
         super(Error, self).__init__(exit_code=exit_code, message=message)
 
 
+if TYPE_CHECKING:
+    _T = TypeVar("_T")
+
+
+@attr.s(frozen=True)
+class ResultError(Exception):
+    """Wraps an Error in an exception for use in short-circuiting via `try_` / `catch`."""
+
+    error = attr.ib()  # type: Error
+
+
+def try_(result):
+    # type: (Union[_T, Error]) -> _T
```

Comment:
How about calling it `unwrap()` from Rust? Seems more descriptive and avoids confusion with the std lib `try`.

Also you could avoid the coupling with `catch` in the docstring - it's often helpful to combine `catch()` and `unwrap()`, but it's not necessary.

Reply:
The `try_` is actually from rust! Before `?` there was only the `try!` macro.

Reply:
And this unwrap does nothing like the panic a non-Ok Result unwrap does. I'll think on alternative names.

Reply:
As one unfamiliar with rust, I don't see the connection between them wrap and catch.. I think try_ makes sense, as it models a similar behavior as the standard try/catch.